### PR TITLE
finished post req for category input

### PR DESCRIPTION
**Pull request by Warrenwwang1**
fixed issue #57

### CODE REVIEW

File: client/src/components/Organizer/CategoryInput/CategoryInput.jsx [hunk 1]
```diff
@@ -56,61 +58,92 @@ class CategoryInput extends Component {
...
     const res = await window.fetch('/api/lists', {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json'
       },
       body: JSON.stringify({
-        apis: this.state.apis,
-        general_categories: this.state.general_categories,
-        fellowships: this.state.fellowships
+        // apis: this.state.apis,
```
Comment (partyshah) [inline client/src/components/Organizer/CategoryInput/CategoryInput.jsx]: delete comments

File: client/src/components/Organizer/CategoryInput/CategoryInput.jsx [hunk 2]
```diff
@@ -56,61 +58,92 @@ class CategoryInput extends Component {
...
     this.setState((prevState) => {
       const awards = prevState.general_categories.slice()
+      var item = awards[index];
+
       awards.splice(index, 1)
-      return {general_categories: awards} 
+      return {
+        general_categories: awards,
+        deleted: prevState.deleted.concat([['general', item]])
+
```
Comment (lajiang2) [inline client/src/components/Organizer/CategoryInput/CategoryInput.jsx]: get rid of this

File: client/src/components/Organizer/CategoryInput/CategoryInput.jsx (hunk 1 above)
Comment (lajiang2) [inline client/src/components/Organizer/CategoryInput/CategoryInput.jsx]: delete this

**Review by partyshah — CHANGES_REQUESTED** [review]

File: client/src/components/Organizer/CategoryInput/CategoryInput.jsx [hunk 3]
```diff
@@ -56,61 +58,92 @@ class CategoryInput extends Component {
...
+        fellowships: prevState.fellowships.concat(fellowship),
+        added: prevState.added.concat([['fellowships', fellowship]])
+      }
     });
   }
 
   removeAPIItem(index) {
     this.setState((prevState) => {
       const awards = prevState.apis.slice();
+      var item = awards[index];
```
Comment (lajiang2) [inline client/src/components/Organizer/CategoryInput/CategoryInput.jsx]: don't use var

**Review by lajiang2 — APPROVED** [review]
looks good, small changes

**Review by partyshah — APPROVED** [review]

**Pull request merged**